Update build-deploy workflow for outdated actions

fredvd opened the pull request:
* Remove matrix variables because there is (no longer?) a matrix configuration
* Update setup-python to remove node 12 warning, use built-in cache 
* update setup-node, user built-in cache support
* set yarn lock file location
* Update action-rsyncer  to 2.0.0

stevepiercy committed:
Overhaul Contributing to Documentation
- Start following Microsoft Style Guide, specifically use Sentence case for headings, not Title Case.
- Rename and refactor docs-writing-guide.md to myst-reference.md, because that is what it really is.
- Break out Sphinx extensions into its own page.
- Reorganize sections into a more logical order.
- Add some Plone-specific documentation tips.

fredvd committed:
Update workflows:
Remove matrix variables because there is (no longer?) a matrix configuration
Update setup-python to remove node 12 warning, use built-in cache
update setup-node, user built-in cache support
set yarn lock file location
Update action-rsyncer  to 2.0.0

fredvd committed:
Update workflows:
Remove matrix variables because there is (no longer?) a matrix configuration
Update setup-python to remove node 12 warning, use built-in cache
update setup-node, user built-in cache support
set yarn lock file location
Update action-rsyncer  to 2.0.0

fredvd committed:
Update build_deploy.yml

Add our own banch temporarily so I can actually test the changes

fredvd commented:
The updated workflow seems to run and deploy fine, but the advertised now builtin caching support in the latest setup-node and setup-python versions seem to interact.  The python built is cached, but the node one isn't, and if you check the output of the post actions where the cache is saved you can see why::

https://github.com/plone/documentation/actions/runs/4158053094/jobs/7192998699

<img width="1890" alt="build-deploy-post" src="https://user-images.githubusercontent.com/394784/218335458-48f5d91f-4179-46f2-9232-5fff71e9d39d.png">

The setup-node cache is 0 mb.  So I'll either need to look up hopefully existing parameters to point to the right directory, or go back to manual cache actions

fredvd commented:
Hmm.  I'll look into the cache action tomorrow.  Apparently the out of the box cache integration in setup-node for now only works if the project is in the root,  other locations is 'work in progress' with weird workarounds: 

https://github.com/actions/setup-node/blob/main/docs/adrs/0000-caching-dependencies.md#goals--anti-goals

https://github.com/actions/setup-node/issues/531

fredvd commented:
@stevepiercy @ericof updating the build-deploy on this branch/pr

fredvd commented:
Sorry, I better ping @sneridagh here.   

Now that I’ve seen this workflow, and want to fix cache support, why are storybook build/deploy and the docs deploy in one workflow? 

Isn’t it better to separate them?  Do we keep different storybooks per released volto or docunentation releases?

stevepiercy commented on .github/workflows/build_deploy.yml:
```diff
@@ -4,18 +4,18 @@ on:
   push:
     branches:
       - "6-dev"
+      - "update_workflows"
```
Should we use this branch for updating workflows in the future, or is it just a one-time thing and should be removed before merging?

fredvd replied:
this is just temporarily because otherwise the worklow doesn't run and the first real test will be when it gets merged to main/master.  Like what happened with our updated set variable line.

stevepiercy reviewed (dismissed):
LGTM, pending resolution of merge conflict, and review from AI team.

stevepiercy commented:
> Now that I’ve seen this workflow, and want to fix cache support, why are storybook build/deploy and the docs deploy in one workflow?

I'm sorry, I don't follow. Can you point out where you see this, so I understand what you mean?

> Isn’t it better to separate them? Do we keep different storybooks per released volto or docunentation releases?

We do not maintain separate versions of storybook. Perhaps we should? That's more of a Volto internal question.

Originally the idea was to move the storybook from `voltocms.com` to `docs.plone.org`. It was piggy-backed onto deploying docs to give it a home under `plone.org`. These days, I think it makes sense to pull the storybook out of `6.docs.plone.org` and into `#.storybook.plone.org`, where it can keep its versioning aligned with Volto. It would definitely speed up docs deployment.

stevepiercy committed:
Reword per https://github.com/plone/documentation/pull/1404#discussion_r1099892664

stevepiercy committed:
Insert "for how" to avoid stumbling, per https://github.com/plone/documentation/pull/1404#discussion_r1099896548

stevepiercy committed:
Add sections for MyST extension `html_image` and videos

stevepiercy committed:
Fix link

sneridagh commented:
> Sorry, I better ping @sneridagh here.
> 
> Now that I’ve seen this workflow, and want to fix cache support, why are storybook build/deploy and the docs deploy in one workflow?
> 
> Isn’t it better to separate them? Do we keep different storybooks per released volto or docunentation releases?

@fredvd for now it's ok to keep them together. In the future if the storybooks for different versions really differ, we can consider doing it.

sneridagh commented:
@fredvd @stevepiercy reading again, just to be more clear. I meant together, as not maintain several versions of the storybook build. Separate them in different build jobs is fine for me!

stevepiercy committed:
Update submodules/plone.restapi submodules/volto tips

stevepiercy committed:
Add Python as a pre-requisite
Make spaces consistent in Makefile

stevepiercy committed:
Add icon to documentation repo to remove warning.
Set notfound_urls_prefix to empty string to remove warning.

stevepiercy committed:
Add comment to refer to https://github.com/plone/documentation/pull/1442

fredvd committed:
Bring back node caching

The builtin caching of the setup-node action does not support nodejs installs in sub directories
Fix the set-output though

fredvd committed:
Bring back node caching

The builtin caching of the setup-node action does not support nodejs installs in sub directories
Fix the set-output though

fredvd commented:
Created issuse #1450 to continue/store the discussion on splitting storybook. 

Closing this PR/branch and creating a new one with only the workflow updates.  I did a rebase on 6-dev but I don't trust myself, why is my branch suddenly 14 commits behind 6-dev....

fredvd commented:
New PR: #1451
